### ISSUE TITLE
Pixel 6a / Android 12 crashes when trying to open OpenHab 

### ISSUE THREAD

**Issue by M1Joe** [openhab/openhab-android#2991]
### Actual behaviour
The app immediately crashes on my pixel 6a (running Android 12) whenever I try to open it. This is a new phone with the latest version of openHab installed from the play store (2.20.0)

### Expected behaviour
The app should open normally.

 
### Steps to reproduce
1. Download the app on a pixel 6a.
2. Attempt to run the app and see that the app won't open.

### Can you reproduce the issue in demo mode?
I can't get into demo model.

#### Client
* Android version: 12
* Device model: Pixel 6a
* App version 2.20.0
* Build flavor : Downloaded from play store
* Device language: English

**Comment by mueller-ma** [comment]
For this combination of openHAB app version, device and Android version I see a single crash:

````
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6767)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:256)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2091)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:7870)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1003)
Caused by: com.google.crypto.tink.shaded.protobuf.InvalidProtocolBufferException: 
  at com.google.crypto.tink.shaded.protobuf.GeneratedMessageLite.parsePartialFrom (GeneratedMessageLite.java:1566)
  at com.google.crypto.tink.shaded.protobuf.GeneratedMessageLite.parseFrom (GeneratedMessageLite.java:1664)
  at com.google.crypto.tink.proto.Keyset.parseFrom (Keyset.java:957)
  at com.google.crypto.tink.integration.android.SharedPrefKeysetReader.read (SharedPrefKeysetReader.java:84)
  at com.google.crypto.tink.CleartextKeysetHandle.read (CleartextKeysetHandle.java:58)
  at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.read (AndroidKeysetManager.java:328)
  at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.readOrGenerateNewKeyset (AndroidKeysetManager.java:287)
  at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.build (AndroidKeysetManager.java:238)
  at androidx.security.crypto.EncryptedSharedPreferences.create (EncryptedSharedPreferences.java:123)
  at org.openhab.habdroid.core.OpenHabApplication.getEncryptedSharedPrefs (OpenHabApplication.kt:72)
  at org.openhab.habdroid.core.OpenHabApplication.access$getEncryptedSharedPrefs (OpenHabApplication.kt:44)
  at org.openhab.habdroid.core.OpenHabApplication$secretPrefs$2.invoke (OpenHabApplication.kt:52)
  at org.openhab.habdroid.core.OpenHabApplication$secretPrefs$2.invoke (OpenHabApplication.kt:49)
  at kotlin.SynchronizedLazyImpl.getValue (LazyJVM.kt:74)
  at org.openhab.habdroid.core.OpenHabApplication.getSecretPrefs (OpenHabApplication.kt:49)
  at org.openhab.habdroid.util.ExtensionFuncsKt.getSecretPrefs (ExtensionFuncs.kt:289)
  at org.openhab.habdroid.core.connection.ConnectionFactory$Companion.initialize (ConnectionFactory.kt:523)
  at org.openhab.habdroid.core.OpenHabApplication.onCreate (OpenHabApplication.kt:90)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1223)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6762)
````

**Comment by itb3** [comment]
I had the same issue, OpenHab crashes on startup, same exception. Brand new Pixel 6 Pro (Android 13), OpenHab installed automatically by the standard Android phone migration tool. A workaround is to clear the cache, which was also migrated. You'll loose your settings, which was not a big issue in my case.

**Comment by mueller-ma** [comment]
Internally openHAB has two preference files: One encrypted one for credentials and one for all other settings. The encryption is done by a key that cannot be transfered to a new device, so the backup for credentials is disabled.

However with Android 12 a new file is required to specify which files has to be included in the backup. Because it's currently missing, both preference files are backed up and restored. The new phone doesn't have the key to decrypt the encrypted file and therefore crashes.
Should be fixed for new backups with #3018.
